crates/jql-parser/src/combinators.rs

```diff
@@ -49,232 +64,182 @@ static TRUNCATE: &str = "!";
 /// A combinator which takes an `inner` parser and produces a parser which also
 /// consumes both leading and trailing whitespaces, returning the output of
 /// `inner`.
-pub(crate) fn trim<'a, F, O, E>(inner: F) -> impl FnMut(&'a str) -> IResult<&'a str, O, E>
+pub(crate) fn trim<'a, F, O, E>(inner: F) -> impl Parser<&'a str, O, E>
 where
-    E: ParseError<&'a str>,
-    F: FnMut(&'a str) -> IResult<&'a str, O, E>,
+    E: ParserError<&'a str>,
+    F: Parser<&'a str, O, E>,
 {
     delimited(multispace0, inner, multispace0)
 }
 
 /// A combinator which parses a stringified number as an `Index`.
-pub(crate) fn parse_number(input: &str) -> IResult<&str, Index> {
-    map_res(recognize(digit1), |index: &str| {
-        index.parse::<usize>().map(Index)
-    })(input)
+pub(crate) fn parse_number(input: &mut &str) -> PResult<Index> {
+    digit1
+        .try_map(|index: &str| index.parse::<usize>().map(Index))
+        .parse_next(input)
 }
 
 /// A combinator which parses a key surrounded by double quotes.
-pub(crate) fn parse_key<'a, E>() -> impl FnMut(&'a str) -> IResult<&'a str, &'a str, E>
-where
-    E: ParseError<&'a str>,
-{
-    trim(delimited(char('"'), take_until(r#"""#), char('"')))
+pub(crate) fn parse_key<'a>(input: &mut &'a str) -> PResult<&'a str> {
+    trim(delimited(DOUBLE_QUOTE, take_until0(r#"""#), DOUBLE_QUOTE)).parse_next(input)
```

Comment:
Since you use `take_until*`, running `cargo add winnow -F simd` will re-enable optimizations that we made disabled by-default compared to `nom`

Reply:
Good to know, thanks!

Reply:
0.5.22 is now out with the fix

Reply:
That was fast! Thanks a ton.